vsintegration/src/FSharp.Editor/FindReferencesService.fs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Collections.Immutable
+open System.Composition
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Host.Mef
+open Microsoft.CodeAnalysis.Editor
+open Microsoft.CodeAnalysis.Editor.Host
+open Microsoft.CodeAnalysis.Navigation
+open Microsoft.CodeAnalysis.FindSymbols
+open Microsoft.CodeAnalysis.FindReferences
+
+open Microsoft.VisualStudio.FSharp.LanguageService
+
+open Microsoft.FSharp.Compiler.Range
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+[<ExportLanguageService(typeof<IStreamingFindReferencesService>, FSharpCommonConstants.FSharpLanguageName); Shared>]
+type internal FSharpFindReferencesService
+    [<ImportingConstructor>]
+    (
+        checkerProvider: FSharpCheckerProvider,
+        projectInfoManager: ProjectInfoManager
+    ) =
+
+    let findReferencedSymbolsAsync(document: Document, position: int, context: FindReferencesContext) : Async<unit> =
+        async {
+            let! sourceText = document.GetTextAsync(context.CancellationToken) |> Async.AwaitTask
+            let! textVersion = document.GetTextVersionAsync(context.CancellationToken) |> Async.AwaitTask
+            let checker = checkerProvider.Checker
+            let! options = projectInfoManager.TryGetOptionsForDocumentOrProject(document)
+            match options with
+            | Some options ->
+                let! _parseResults, checkResultsAnswer = checker.ParseAndCheckFileInProject(document.FilePath, hash textVersion, sourceText.ToString(), options)
+                let checkFileResults = 
+                    match checkResultsAnswer with
+                    | FSharpCheckFileAnswer.Aborted -> failwith "Compilation isn't complete yet"
```

Comment:
> This logic is used everywhere. 

That's very concerning :)   Many Roslyn extension points are nto hardened to getting a non-cancellation exception.  These exceptions will tear down VS in many cases.

The two approaches i outlined above are far preferred.  

> What about repeat calling ParseAndCheckFileInProject until it returns Succeeded?

If that works, then sure :)   I would recommend doing it with some sort Task.Delay so that you're not consuming a whole core busy waiting.  Also, remember to continually check hte cancellation token.